app/controllers/foreman_discovery/concerns/hosts_controller_extensions.rb

```diff
@@ -0,0 +1,18 @@
+module ForemanDiscovery
+  module Concerns
+    module HostsControllerExtensions
+      extend ActiveSupport::Concern
+      included do
+        before_action :set_discovered_params
+      end
+
+      # Change params to what the hosts controller expects. Certain methods
+      # in the hosts controller like all the _selected methods, taxonomy_scope,
+      # etc.. expect a params[:host] to work.
+      def set_discovered_params
+        return unless request.url.match(/discovered_hosts/).present?
```

Comment:
So my concern is these is an url like `/some/url?param=discovered_hosts` that could match. AFAIR we don't have this one, but we could we.

My proposal is either match only the `path` element or at least add slash: `/\/discovered_hosts\//`.